### PR TITLE
Aggregate sub-component negative condition into HCO conditions.

### PR DESCRIPTION
**Pull request by nunnatsa**
Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -239,84 +206,14 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
...
-				Message: fmt.Sprintf("Error while reconciling: %v", err),
-			})
-			// don't want to overwrite the actual reconcile failure
-			uErr := r.client.Status().Update(context.TODO(), instance)
-			if uErr != nil {
-				reqLogger.Error(uErr, "Failed to update conditions")
-			}
-			return reconcile.Result{}, err
-		}
+	err = r.ensureComponents(instance, request, reqLogger)
```
Comment (djzager) [inline pkg/controller/hyperconverged/hyperconverged_controller.go:214]: If you want to refactor the `Reconcile()` func, you should do that in a separate PR.

**Comment by nunnatsa** [comment]
Done in #564
